src/common/ranking_utils.h

```diff
@@ -215,6 +216,9 @@ class RankingCache {
     if (!info.weights_.Empty()) {
       CHECK_EQ(Groups(), info.weights_.Size()) << error::GroupWeight();
     }
+    if (param_.HasTruncation()) {
```

Comment:
How is this condition related to the use of the ``mean`` method? My understanding is that the normalization factor is ``n_pairs`` for the ``mean`` method and the gradient if other method is used?